release/pkg/versions.go

```diff
@@ -9,18 +11,13 @@ import (
 	"github.com/pkg/errors"
 )
 
-func BuildComponentVersion(versioner projectVersioner) (string, error) {
+func BuildComponentVersion(versioner projectVersioner, componentCheckSum string) (string, error) {
 	patchVersion, err := versioner.patchVersion()
 	if err != nil {
 		return "", err
 	}
 
-	metadata, err := versioner.buildMetadata()
-	if err != nil {
-		return "", err
-	}
-
-	return fmt.Sprintf("%s+%s", patchVersion, metadata), nil
+	return fmt.Sprintf("%s+%s", patchVersion, componentCheckSum), nil
```

Comment:
Future idea:
we could use the checksum just to check if something has changed (as opposed to have it in the semver itself) and use a monotonic int like
```
if oldCheskum != newChecksum
  v1.0.1+build.1 -> v1.0.1+build+2
fi
```

Nothing wrong with using the hash for now, but this will probably be a bit easier to read and it will be more clear if changes between versions (eg. when running `upgrade plan`) are upgrades or downgrades

Reply:
I like this idea. I'll make a note if it. Perhaps I could add this in a separate PR